src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/StubCodeContext.cs

```diff
@@ -42,20 +47,25 @@ public enum Stage
             /// </remarks>
             Invoke,
 
+            /// <summary>
+            /// Capture native values to ensure that we do not leak if an exception is thrown during unmarshalling
+            /// </summary>
+            UnmarshalCapture,
+
             /// <summary>
             /// Convert native data to managed data
             /// </summary>
             Unmarshal,
 
             /// <summary>
-            /// Perform any cleanup required
+            /// Notify a marshaller object that the Invoke stage was successful (no exception thrown).
```

Comment:
The `Invoke` stage is more than just the actual calli, right? If not, we are missing some clarity here because exceptions are only accepted from an unmanaged calli on Windows.